operators/multiclusterobservability/controllers/placementrule/role.go

```diff
@@ -362,7 +362,8 @@ func deleteDeprecatedRoles(c client.Client) {
 		log.Error(err, "Failed to list deprecated roles")
 		return
 	}
-	for _, role := range roleList.Items {
```

Comment:
What is wrong here?

Reply:
Implicit memory aliasing in for loop. for example: https://husni.dev/beware-of-implicit-memory-aliasing-in-go-foor-loop/

Reply:
refer to https://sonarcloud.io/project/issues?id=open-cluster-management_multicluster-observability-operator&issues=AXrq11lHBBOe0YXDjbrG&open=AXrq11lHBBOe0YXDjbrG